### PR TITLE
{AKS} Fix failed live test case and add custom virtual network preparer

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

* Fix the “test_aks_create_default_service_with_virtual_node_addon” test case in ACS module which would fail in live test.
  * Provide the same location as the resource group preparer
  * Provide a subnet address prefix to avoid conflicts with default AKS cluster cidr
* Add a custom "AKSCustomVirtualNetworkPreparer" to replace class "VirtualNetworkPreparer" to meet the need to provide the above required resources

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py [hunk 1]
```diff
@@ -31,3 +32,87 @@ def __init__(
...
+                    'az network vnet delete --name {} --resource-group {}'.
+                    format(name, self._get_resource_group(**kwargs)))
+            except CloudError:
+                # deprecated since the SDK that network commands relies on has been migrated to track 2
+                # deletion of vnet may fail as service could create subresources like IPConfig. We could rely on the deletion of resource group to delete the vnet.
+                pass
+            except HttpResponseError as ex:
+                # new excetption class adopted by track 2 SDK
+                # network resources are still used by other resources such as vm, and there is no clean way to delete these dependencies, rely on delete rg.
+                if "(InUseSubnetCannotBeDeleted)" in ex.message:
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py:110]: is there any error code we can use instead of the text matching?

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py [hunk 2]
```diff
@@ -31,3 +32,87 @@ def __init__(
...
+        self._update_address_prefixes(**kwargs)
+        self.live_only_execute(
+            self.cli_ctx,
+            template.format(self._get_resource_group(**kwargs), self.location,
+                            name, self.address_prefixes))
+
+        self.test_class_instance.kwargs[self.key] = name
+        return {self.parameter_name: name}
+
+    def remove_resource(self, name, **kwargs):
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py:94]: if this always fail, why we ever need to call this?

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py (hunk 2 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py:119]: set to pass and leave comments in commit [a7e9704](https://github.com/Azure/azure-cli/pull/18168/commits/a7e97046c8cdd779ef0fb717427ca25fffbca3bb).

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py]: change to pass and leave comments in commit [a7e9704](https://github.com/Azure/azure-cli/pull/18168/commits/a7e97046c8cdd779ef0fb717427ca25fffbca3bb).

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py [hunk 3]
```diff
@@ -3,31 +3,130 @@
...
-            dev_setting_name='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_NAME',
-            dev_setting_location='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_LOCATION',
-            random_name_length=75,
-            key='rg'):
-        super(AKSCustomResourceGroupPreparer,
-              self).__init__(name_prefix, parameter_name,
-                             parameter_name_for_location, location,
-                             dev_setting_name, dev_setting_location,
-                             random_name_length, key)
+        self,
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py:19]: I think this will have conflict with your other pr.

**Review by andyliuliming — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py (hunk 3 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py:19]: changed due to strict style/format check, I will pay more attention to other prs.

**Review by Juliehzl — APPROVED** [review]
LGTM

**Pull request merged**